File: src/main/java/com/company/erp/features/employees/entity/Employee.java
package com.company.erp.features.employees.entity;

import com.company.erp.core.database.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "employees")
public class Employee extends BaseEntity {

    /** Optional FK to the login user. One employee per user (unique on the DB side). */
    @Column(name = "user_id")
    private Long userId;

    @Column(name = "employee_no", nullable = false, unique = true)
    private String employeeNo;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "work_email")
    private String workEmail;

    @Column
    private String phone;

    @Column
    private String position;

    @Column
    private String department;

    @Column(name = "hire_date")
    private LocalDate hireDate;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    @Column
    private String gender;

    @Column
    private String address;

    @Column(name = "avatar_url")
    private String avatarUrl;

    @Column(name = "avatar_content_type")
    private String avatarContentType;

    @Column(name = "avatar_uploaded_at")
    private Instant avatarUploadedAt;

    @Column(name = "emergency_contact")
    private String emergencyContact;

    @Column(name = "emergency_phone")
    private String emergencyPhone;

    @Column(name = "last_login_at")
    private Instant lastLoginAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EmployeeStatus status = EmployeeStatus.ACTIVE;
}

File: app/Features/Employees/Models/Employee.php
<?php

namespace App\Features\Employees\Models;

use App\Support\Database\BlamesUser;
use Illuminate\Database\Eloquent\Model;

/**
 * HR-side employee profile. May link to a login user (user_id, unique).
 *
 * @property int $id
 * @property int|null $user_id
 * @property string $employee_no
 * @property string $full_name
 * @property string|null $work_email
 * @property string|null $phone
 * @property string|null $position
 * @property string|null $department
 * @property \Illuminate\Support\Carbon|null $hire_date
 * @property \Illuminate\Support\Carbon|null $date_of_birth
 * @property string|null $gender
 * @property string|null $address
 * @property string|null $avatar_url
 * @property string|null $avatar_content_type
 * @property \Illuminate\Support\Carbon|null $avatar_uploaded_at
 * @property string|null $emergency_contact
 * @property string|null $emergency_phone
 * @property \Illuminate\Support\Carbon|null $last_login_at
 * @property string $status
 */
class Employee extends Model
{
    use BlamesUser;

    protected $table = 'employees';

    protected $fillable = [
        'user_id', 'employee_no', 'full_name', 'work_email', 'phone', 'position',
        'department', 'hire_date', 'date_of_birth', 'gender', 'address', 'avatar_url',
        'avatar_content_type', 'avatar_uploaded_at', 'emergency_contact', 'emergency_phone',
        'last_login_at', 'status',
    ];

    protected $casts = [
        'hire_date' => 'date',
        'date_of_birth' => 'date',
        'avatar_uploaded_at' => 'datetime',
        'last_login_at' => 'datetime',
    ];

    protected $attributes = [
        'status' => EmployeeStatus::ACTIVE,
    ];
}
